### PR TITLE
feat: throw error on invalid syntax

### LINKED ISSUE #9: Return error when any issue is encountered when parsing
Right now ANTLR does soft fails (it prints the errors to STDERR, but still attempts to parse)

Screenshot 2023-07-27 at 12 08 54 PM

### PR DESCRIPTION
**Pull request by rhamzeh**
## References

- closes #9 

### Sources
- https://www.antlr.org/api/Java/org/antlr/v4/runtime/ANTLRErrorListener.html
- https://stackoverflow.com/questions/18132078/handling-errors-in-antlr4?rq=4
- https://stackoverflow.com/questions/20828864/accumulating-collecting-errors-via-errorlistener-to-handle-after-the-parse

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/go/Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ clean:
 	go clean
 
 test:
-	go test ./...
+	go test ./... -count=1
```
Comment (adriantam) [inline pkg/go/Makefile:13]: why do you need count=1?

File: pkg/go/transformer/dsltojson_test.go [hunk 2]
```diff
@@ -37,4 +39,21 @@ func TestDSLToJSONTransformer(t *testing.T) {
 			require.JSONEq(t, string(jsonBytes), string(bytes))
 		})
 	}
+
+	testCases2, err := LoadInvalidDslSyntaxTestCases()
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson_test.go:43]: Will it make sense to break it up into two different function?

File: pkg/go/transformer/dsltojson_test.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 func TestDSLToJSONTransformer(t *testing.T) {
-	testCases, err := LoadTransformerTestCases()
+	testCases, err := LoadValidTransformerTestCases()
 	if err != nil {
 		log.Fatal(err)
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson_test.go:16]: Since this is test, why not t.Fatal(err)?

File: pkg/go/transformer/dsltojson_test.go [hunk 4]
```diff
@@ -37,4 +39,21 @@ func TestDSLToJSONTransformer(t *testing.T) {
 			require.JSONEq(t, string(jsonBytes), string(bytes))
 		})
 	}
+
+	testCases2, err := LoadInvalidDslSyntaxTestCases()
+	if err != nil {
+		log.Fatal(err)
```
Comment (adriantam) [inline pkg/go/transformer/dsltojson_test.go:45]: instead of log.Fatal(err), use t.Fatal(err)

File: pkg/go/transformer/jsontodsl_test.go [hunk 5]
```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func TestJSONToDSLTransformer(t *testing.T) {
-	testCases, err := LoadTransformerTestCases()
+	testCases, err := LoadValidTransformerTestCases()
 	if err != nil {
 		log.Fatal(err)
```
Comment (adriantam) [inline pkg/go/transformer/jsontodsl_test.go:14]: use t.Fatal if possible

File: pkg/go/transformer/testcases_test.go [hunk 6]
```diff
@@ -1,34 +1,31 @@
...
 	"os"
 	"path/filepath"
 )
 
-type TestCases struct {
-	Cases []TestCase
-}
-
-type TestCase struct {
+type ValidTestCase struct {
```
Comment (adriantam) [inline pkg/go/transformer/testcases_test.go:9]: Are we exporting these?  If not, why capital on struct and function names?

File: pkg/go/Makefile (hunk 1 above)
Comment (rhamzeh) [reply pkg/go/Makefile:13]: To disable the cache, otherwise rerunning tests after modifying the test data would only run the previous cached tests

**Pull request merged**